Refactor docs

waylan opened the pull request:
This is the beginnings of the work summarized in #1629. A work in progress.

oprypin commented on mkdocs.yml:
```diff
@@ -15,13 +15,19 @@ theme:
         - Configuration: user-guide/configuration.md
         - Deploying Your Docs: user-guide/deploying-your-docs.md
-        - Custom Themes: user-guide/custom-themes.md
-        - Plugins: user-guide/plugins.md
```
Please don't break URLs, there are many of them in the wild.
Some options would be:
* Don't change filenames despite changing sections.
* Use "redirects" plugin.

waylan replied:
These are not `user-guide` material. They never should have been put in that directory. Therefore, they are being moved. Redirects could be an option.

waylan replied:
Redirects have been defined using the plugin. Thanks for the suggestion.

waylan commented:
I redesigned the homepage. First I moved the "Installation" and "Getting Started" to their own respective pages. Then, as the homepage had no TOC to speak of, I removed it and styled the list of features as a grid of cards. See the screenshots below for the large, medium, and small sizes. Any and all feedback is welcome. 

# Large

![homepage_large](https://user-images.githubusercontent.com/78846/113432983-597db280-93ac-11eb-89fa-6389d086dc76.png)

# Medium

![homepage_medium](https://user-images.githubusercontent.com/78846/113433001-5f739380-93ac-11eb-8e6a-449317e1e8cc.png)

# Small

![homepage_small](https://user-images.githubusercontent.com/78846/113433018-6b5f5580-93ac-11eb-89bb-f1ef782edf84.png)

waylan commented:
I should mention a few things:

1. These changes are only for the `mkdocs.org` website (our documentation). There have been no changes made to the theme. I used Bootstrap markup via raw HTML in Markdown and added a few custom CSS rules.
2. The changes in appearance are only to the homepage. The design of all other pages is unchanged (they still have their TOC, etc).

waylan commented:
This doesn't address everything in #1629 (most notably the API docs), but I think this is as far as I'm taking this now. The refactor is done. Everything else are additions, which can be added at any time.
